### PR TITLE
feat(KubernetesPodOperator): Add support of container_security_context

### PR DESCRIPTION
**Pull request by yyvess**
Allow to define a container security context on KubernetesPodOperator.

Why: 
On clusters restricted with strong security policy, pods cannot be executed without disable privilege escalation.

About test :
I successfully run locally there tests : _tests/providers/cncf/kubernetes/operators/test_kubernetes_pod.py_
But I was not able to run tests _kubernetes_tests/test_kubernetes_pod_operator.py_ :-(

### CODE REVIEW

File: kubernetes_tests/test_kubernetes_pod_operator.py [hunk 1]
```diff
@@ -495,12 +513,11 @@ def test_run_as_user_root(self):
...
 
-    def test_run_as_user_non_root(self):
+    def test_disable_privilege_escalation(self):
         security_context = {
-            'securityContext': {
-                'runAsUser': 1000,
-            }
+            'runAsUser': 1000,
         }
+        container_security_context = {'allowPrivilegeEscalation': false}
```
Comment (uranusjr) [inline kubernetes_tests/test_kubernetes_pod_operator.py]:
```suggestion
        container_security_context = {'allowPrivilegeEscalation': False}
```

?

File: kubernetes_tests/test_kubernetes_pod_operator.py [hunk 2]
```diff
@@ -472,10 +472,28 @@ def test_volume_mount(self):
...
+        actual_pod = self.api_client.sanitize_for_serialization(k.pod)
+        self.expected_pod['spec']['securityContext'] = security_context
+        assert self.expected_pod == actual_pod
+
+    def test_run_as_user_non_root(self):
         security_context = {
-            'securityContext': {
-                'runAsUser': 0,
-            }
+            'runAsUser': 1000,
```
Comment (uranusjr) [inline kubernetes_tests/test_kubernetes_pod_operator.py]: Is this related, and why?

Comment (potiuk) [reply kubernetes_tests/test_kubernetes_pod_operator.py]: And this looks like a breaking change. Why do we change it ?

**Comment by yyvess** [comment]
@uranusjr please ☀️

**Comment by uranusjr** [comment]
You didn’t answer why you changed the `security_context` values in tests and how it affects compatibility.

**Comment by yyvess** [comment]
> You didn’t answer why you changed the `security_context` values in tests and how it affects compatibility.

@uranusjr I was answed you ..... 
Changing only tests cannot affect compatibility 🙄
And as you can see all tests still green 

I re-copy here my previous answer, please read it =>

@uranusjr Why => Because for me that was incorrect but isn't related to my main change as I write previously.
@potiuk I don't thinks that break anythings, the question is : why that test was not failed before my change ...

The variable security_context is injected on k8s.V1PodSpec =>
spec=k8s.V1PodSpec(..., security_context=self.security_context,

Look API documentation of K8s and you can see that security_context don't have a property securityContext =>

https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1PodSpec.md
https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1PodSecurityContext.md
It's seem that main issue is how the test is write. I am sure that on this test you can set security_context = {'hello': 'world'} and the test still green.

Perhaps we must change the type of security_context & container_security_context on class KubernetesPodOperator that is actually defined as Dict

security_context: Optional[Dict] = None, container_security_context: Optional[Dict] = None,

Should be more safer to declare it as is

security_context: Optional[k8s.V1PodSecurityContext] = None, container_security_context: Optional[k8s.V1SecurityContext] = None,

But should make on a different PR is you approved to change types

**Comment by uranusjr** [comment]
> It's seem that main issue is how the test is write. I am sure that on this test you can set security_context = {'hello': 'world'} and the test still green.

This is the context I was looking for.

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
